packages/graph/hash_graph/lib/graph/src/shared/identifier/time/projection.rs

```diff
@@ -71,35 +89,63 @@ pub type UnresolvedDecisionTimeImage = UnresolvedImage<DecisionTime>;
 
 impl ToSchema for UnresolvedDecisionTimeImage {
     fn schema() -> openapi::RefOr<openapi::Schema> {
-        openapi::Schema::AllOf(
-            openapi::AllOfBuilder::new()
-                .item(
-                    openapi::ObjectBuilder::new()
-                        .property("axis", openapi::Ref::from_schema_name("DecisionTime"))
-                        .required("axis"),
-                )
-                .item(UnresolvedTimeInterval::<DecisionTime>::schema())
-                .build(),
-        )
-        .into()
+        openapi::ObjectBuilder::new()
+            .property("axis", openapi::Ref::from_schema_name("DecisionTime"))
+            .required("axis")
+            .property(
+                "start",
+                openapi::Schema::OneOf(
+                    openapi::OneOfBuilder::new()
+                        .item(openapi::Ref::from_schema_name("TimeIntervalBound"))
+                        .nullable(true)
+                        .build(),
+                ),
+            )
+            .required("start")
+            .property(
+                "end",
+                openapi::Schema::OneOf(
+                    openapi::OneOfBuilder::new()
+                        .item(openapi::Ref::from_schema_name("TimeIntervalBound"))
+                        .nullable(true)
+                        .build(),
+                ),
+            )
+            .required("end")
+            .build()
+            .into()
     }
 }
 
 pub type UnresolvedTransactionTimeImage = UnresolvedImage<TransactionTime>;
 
 impl ToSchema for UnresolvedTransactionTimeImage {
     fn schema() -> openapi::RefOr<openapi::Schema> {
-        openapi::Schema::AllOf(
-            openapi::AllOfBuilder::new()
-                .item(
-                    openapi::ObjectBuilder::new()
-                        .property("axis", openapi::Ref::from_schema_name("TransactionTime"))
-                        .required("axis"),
-                )
-                .item(UnresolvedTimeInterval::<DecisionTime>::schema())
```

Comment:
Curious about the change made here. Instead of `allOf`-ing the `UnresolvedTimeInterval` we now explicitly define `start` and `end` with nullable `TimeIntervalBound`. Aren't these equivalent or did the schema of `UnresolvedTimeInterval` turn the `Option`s into `undefined`?

Reply:
The `Option` turns it into `undefined`. Also, the `allOf` generates those weird duplicated definitions. I will test if I can tweak this a little bit and report back.